dffml/operation/archive.py

```diff
@@ -0,0 +1,74 @@
+import uuid
+from zipfile import ZipFile
+from pathlib import Path
+from typing import Union, Dict, Any, Optional
+
+from ..df.base import op
+from ..df.types import Definition
+
+
+# definitions
+PATH = Definition(name="path", primitive="Union[Path, str]")
+OPTIONS = Definition(name="options", primitive="Dict[str, Any]")
+
+
+@op(
+    inputs={
+        "input_directory_path": PATH,
+        "output_file_path": PATH,
+        "options": OPTIONS,
+    },
+    outputs={},
+)
+async def make_zip_archive(
+    input_directory_path: Union[Path, str],
+    output_file_path: Optional[Union[Path, str]] = None,
+    **options: Optional[Dict[str, Any]],
```

Comment:
Let's hold off on implementing options for now

```suggestion
```

Reply:
Removed